CHANGELOG.md

```diff
@@ -4,14 +4,38 @@
 `psql` with the `-X` flag to prevent any `.psqlrc` commands from
 accidentally triggering the load of a previous DB version.**
 
-## Unreleased
+## 2.4.0 (2021-07-26)
+This release adds new experimental features since the 2.3 release.
+The experimental features in this release are:
+* APIs for chunk manipulation across data nodes in a distributed
+hypertable setup. This includes the ability to add a data node and move
+chunks to the new data node for cluster rebalancing,  
+* time_bucket_ng function, a newer version of `time_bucket`.This function
+supports years, months, days, hours, minutes and seconds. 
+We welcome feedback on the experimental features.
+
+This release also includes several bugfixes. 
+
+**Experimental Features**
+* #3293 Add timescaledb_experimental schema
+* #3302 Add block_new_chunks and allow_new_chunks API to experimental
+schema. Add chunk based refresh_continuous_aggregate.
+* #3211 Introduce experimental time_bucket_ng function
+* #3366 Allow use of experimental time_bucket_ng function in continuous aggregates
+* #3408 Support for seconds, minutes and hours in time_bucket_ng
 
 **Bugfixes**
 * #3401 Fix segfault for RelOptInfo without fdw_private
 * #3411 Verify compressed chunk validity for compressed path
+* #3416 Fix targetlist names for continuous aggregate views
+* #3434 Remove extension check from relcache invalidation callback
+* #3440 Fix remote_tx_heal_data_node to work with only current database
 
```

Comment:
I see in the commit log of master:
```
* commit c270891a4ecd1cb3a108f2709ed8a827b8a93109
| Author: Sven Klemm <sven@timescale.com>
| Date:   Thu Jul 8 10:06:56 2021 +0200
| 
|     Fix segfault for RelOptInfo without fdw_private
|     
|     In nested function invocations hypertable expansion would not work
|     correctly and a hypertable would not be expanded by timescaledb
|     code but by postgres table inheritance leading to fdw_private
|     not being properly initialized.
|     
|     Fixes #3391
```

Should this be part of the bugfixes?

Some more:
```
* commit 637fe3275bb4d710607ed7d6991983e37aa7af81
| Author: Mats Kindahl <mats@timescale.com>
| Date:   Fri Jul 16 13:57:31 2021 +0200
| 
|     Fix targetlist for cagg views
|     
|     If the names for entries in the targetlist for the direct and partial
|     views of the continuous aggregate does not match the attribute names in
|     the tuple descriptor for the result tuple of the user view an error
|     will be generated. This commit fixes this by setting the targetlist
|     resource names of the columns in the direct and partial view to the
|     corresponding attribute name of the user view relation tuple
|     descriptor.
|     
|     Fixes #3051
|     Fixes #3405
```

```
* commit 0277ed74617b915a3e741a966b05998aa9a4d789
| Author: gayyappan <gayathri@timescale.com>
| Date:   Tue Jul 13 11:43:19 2021 -0400
| 
|     Verify compressed chunk validity for insert path
|     
|     When a insert into a compressed chunk is blocked by a
|     concurrent recompress_chunk, the latter process could move
|     the storage for the compressed chunk. Verify validity of
|     the compressed chunk before proceeding to acquire locks.
|     
|     Fixes #3400
```

```
* commit 3a8f396bc7d014a2722b8113236b34b0eda73eec
| Author: Markos Fountoulakis <markos.fountoulakis.senior@gmail.com>
| Date:   Sat Jul 24 13:20:02 2021 +0300
| 
|     Remove check from relcache invalidation callback
|     
|     The timescaledb extension checking code uses syscache lookups to
|     determine whether the proxy table exists. Doing syscache lookups
|     inside the invalidation callback context was corrupting the syscache.
|     For this reason, we remove this callback.
```
The above Fixes #3188

```
| Author: Dmitry Simonenko <dmitry@timescale.com>
| Date:   Sun Jul 25 17:50:49 2021 +0300
| 
|     Fix remote_txn_heal_data_node() for multiple DB's
|     
|     Since pg_prepared_xacts is shared between databases, the healing
|     function tried to resolve prepared transactions created by other
|     distributed databases.
|     
|     This change makes the healing function to work only with current
|     database.
|     
|     Fix #3433
```

Reply:
@mfundul All these PRs are listed in the changelog. Not clear what you are suggesting is missing from the list.

Reply:
I was looking at issue numbers instead of PR numbers, so that was the source of the confusion.